app/controllers/Prism.scala

```diff
@@ -1,22 +1,32 @@
 package controllers
 
-import agent.CollectorAgent
+import agent.{Accounts, CollectorAgent, LabelAgent}
+import akka.actor.ActorSystem
 import collectors._
+import conf.PrismConfiguration
 
-object Prism {
-  val lazyStartup = conf.PrismConfiguration.accounts.lazyStartup
-  val instanceAgent = new CollectorAgent[Instance](InstanceCollectorSet, lazyStartup)
-  val lambdaAgent = new CollectorAgent[Lambda](LambdaCollectorSet, lazyStartup)
-  val dataAgent = new CollectorAgent[Data](DataCollectorSet, lazyStartup)
-  val securityGroupAgent = new CollectorAgent[SecurityGroup](SecurityGroupCollectorSet, lazyStartup)
-  val imageAgent = new CollectorAgent[Image](ImageCollectorSet, lazyStartup)
-  val launchConfigurationAgent = new CollectorAgent[LaunchConfiguration](LaunchConfigurationCollectorSet, lazyStartup)
-  val serverCertificateAgent = new CollectorAgent[ServerCertificate](ServerCertificateCollectorSet, lazyStartup)
-  val acmCertificateAgent = new CollectorAgent[AcmCertificate](AmazonCertificateCollectorSet, lazyStartup)
-  val route53ZoneAgent = new CollectorAgent[Route53Zone](Route53ZoneCollectorSet, lazyStartup)
-  val elbAgent = new CollectorAgent[LoadBalancer](LoadBalancerCollectorSet, lazyStartup)
-  val bucketAgent = new CollectorAgent[Bucket](BucketCollectorSet, lazyStartup)
-  val reservationAgent = new CollectorAgent[Reservation](ReservationCollectorSet, lazyStartup)
+// TODO: Maybe we should refactor this to be PrismAgents and to not be in the controllers package?
+class Prism(prismConfiguration: PrismConfiguration)(actorSystem: ActorSystem) {
+  val labelAgent = new LabelAgent(actorSystem)
+  val accounts = new Accounts(prismConfiguration)
+
+  val lazyStartup: Boolean = prismConfiguration.accounts.lazyStartup
+
+  // TODO: Maybe we should refactor this to not require a circular reference / pass in this
```

Comment:
I went down a bit of a rabbit hole trying to model types that depended on other types. This would actually be easier if the data was all in a DB I expect. Fortunately it didn't get much further than this one example. Bet no one uses it.